app/models/spree/gateway/flow_io.rb

```diff
@@ -90,18 +92,26 @@ def request_refund_store_result(order, amount)
                                                            currency: order.currency)
         response = FlowcommerceSpree.client.refunds.post(FlowcommerceSpree::ORGANIZATION, refund_form)
         response_status = response.status.value
-        if response_status == REFUND_SUCCESS
+        if REFUND_VALID_STATES.include? response_status
           add_refund_to_order(response, order)
+          schedule_status_check(response, order)
           ActiveMerchant::Billing::Response.new(true,
                                                 REFUND_SUCCESS,
                                                 response.to_hash,
                                                 authorization: response.authorization.id)
         else
-          msg = "Partial refund fail. Details: #{response_status}"
-          ActiveMerchant::Billing::Response.new(false, msg, {}, {})
+          ActiveMerchant::Billing::Response.new(false, "Partial refund fail. Details: #{response_status}", {}, {})
         end
       end
 
+      def schedule_status_check(response, order)
+        return if response.status.value != REFUND_PENDING
+
+        Rails.logger.warn("[!] #{self.class} refund request without succeeded status.
+        Order: #{order.id}, refund capture status: #{response.captures.first.capture.status.value}")
```

Comment:
Is it ok to log only the `first`?

Reply:
Good question. To be honest, I am not sure. We can probably improve this and iterate through all associated caputres. In the tests we did there was always one capture though. I think that we can even probably avoid that part of the log and only leave the first part where we store the order's id (I would change that to order's number probably for easy search).

So you know, the capture associated to the refund's transaction is always "succeeded" (not sure why, Flow only confirmed that it was ok that it is shown that way) :man_shrugging: . I will update this message before merging